<?xml version="1.0"?>
<launch>

  <arg name="vectorfield_a" />
  <arg name="vectorfield_b" />
  <arg name="image_width" default="1000"/>
  <arg name="vectorfield_ab" />
  <arg name="as_gridmap" default="False"/>
  <arg name="meter_per_pixel" value="0.003"/>

  <!-- merge vector field a and b -->
  <node name="vf_merger_$(arg vectorfield_a)_$(arg vectorfield_b)" pkg="vector_field_navigation" type="vectorfield_merger_node" output="screen" cwd="ROS_HOME">
    <param name="field1_listener_topic" value="/vectorfield/$(arg vectorfield_a)"/>
    <param name="field2_listener_topic" value="/vectorfield/$(arg vectorfield_b)"/>
    <param name="field_width" value="$(arg image_width)"/>
    <param name="field_height" value="$(arg image_height)"/>
    <param name="vectorfield_publisher_topic" value="$(arg vectorfield_ab"/>
    <param name="normalize" value="1"/>
  </node>

  <group if="$(arg as_gridmap)">
    <node name="vfield_to_gm_$(arg vectorfield_ab" pkg="vector_field_navigation" type="vectorfield_to_gridmap_node" output="screen" cwd="ROS_HOME">
      <param name="meter_per_pixel" value="$(arg meter_per_pixel)"/>
      <param name="frame_id" value="world"/>
      <param name="vectorfield_listener_topic" value="/vectorfield/$(arg vectorfield_ab)"/>
      <param name="gridmap_publisher_topic" value="/vectorfield/$(arg vectorfield_ab)/gridmap"/>
    </node>
  <group/>
</launch>